mindsdb/api/executor/sql_query/steps/insert_step.py

```diff
@@ -30,12 +31,32 @@ def call(self, step):
             if step.is_replace:
                 is_replace = True
 
+        if len(step.table.parts) > 1:
+            integration_name = step.table.parts[0]
+            table_name = Identifier(parts=step.table.parts[1:])
+        else:
+            integration_name = self.context['database']
+            table_name = step.table
+
+        dn = self.session.datahub.get(integration_name)
+
+        if hasattr(dn, 'create_table') is False:
+            raise NotSupportedYet(f"Creating table in '{integration_name}' is not supported")
+
         if step.dataframe is not None:
             data = self.steps_data[step.dataframe.result.step_num]
         elif step.query is not None:
             data = ResultSet()
-            for col in step.query.columns:
-                data.add_column(Column(col.name))
+            if step.query.columns is None:
+                # Is query like: INSERT INTO table VALUES (...)
+                table_columns_df = dn.get_table_columns_df(str(table_name))
+                columns_names = table_columns_df[INF_SCHEMA_COLUMNS_NAMES.COLUMN_NAME].to_list()
```

Comment:
will it work if count of columns in table is equal count of columns in query?

Reply:
if not there should be this exception
https://github.com/mindsdb/mindsdb/blob/d6d7ce3df498e9754ce6fc15d6d01e17523dff23/mindsdb/api/executor/sql_query/result_set.py#L245